sdcm/provision/aws/utils.py

```diff
@@ -251,32 +251,33 @@ def sort_by_index(item: dict) -> str:
 
 
 def network_config_ipv6_workaround_script():
-    return dedent("""
+    return dedent(r"""
         if grep -qi "ubuntu" /etc/os-release; then
             echo "On Ubuntu we don't need this workaround, so done"
         else
             BASE_EC2_NETWORK_URL=http://169.254.169.254/latest/meta-data/network/interfaces/macs/
             MAC=`curl -s ${BASE_EC2_NETWORK_URL}`
             IPv6_CIDR=`curl -s ${BASE_EC2_NETWORK_URL}${MAC}/subnet-ipv6-cidr-blocks`
 
-            while ! ls /etc/sysconfig/network-scripts/ifcfg-eth0; do sleep 1; done
+            NETWORK_DEVICE=`ip -4 route ls | grep default | grep -Po '(?<=dev )(\S+)'`
```

Comment:
on my laptop this returns 2 devices (second one is wifi card). Is there possibility we get more than one device here?

Reply:
maybe, but only one of them should be default gateway
In our cloud machine we have only one network interface (only on specific use case, we have 2 interfaces, and we don't use ipv6 there, at least not yet, if we'll do, we'll need to adapt the script to configure both)

anyhow this is only for older distros like centos7, on ubuntu we don't need all this tricks, ipv6 works out of the box.

the place it didn't work out of the box is in other monitor, that we run ontop of centos7